tests/baselines/reference/declarationEmitPathMappingMonorepo.js

```diff
@@ -0,0 +1,30 @@
+//// [tests/cases/compiler/declarationEmitPathMappingMonorepo.ts] ////
+
+//// [index.d.ts]
+declare module "@ts-bug/a" {
+    export type AText = {
+      value: string;
+    };
+    export function a(text: string): AText;
+  }
+  
+//// [index.ts]
+import { a } from "@ts-bug/a";
+
+export function b(text: string) {
+  return a(text);
+}
+
+//// [index.js]
+"use strict";
+exports.__esModule = true;
+var a_1 = require("@ts-bug/a");
+function b(text) {
+    return a_1.a(text);
+}
+exports.b = b;
+
+
+//// [index.d.ts]
+/// <reference types="@ts-bug/a" />
+export declare function b(text: string): import("@ts-bug/a").AText;

```

Comment:
As an aside, since the import is `import("@ts-bug/a")`, we don't _need_ the `/// <reference types="@ts-bug/a" />`, since they refer to the "same" thing (or at least cause the same file to be found and included). However since the import could be something like `@ts-bug/a/b` (which could be an ambient module in `@ts-bug/a`), we can't generally elide the reference. A cleanup pass where we remove `types` directives which exactly match an `import` somewhere in the file may be pertinent.